core/src/test/java/io/questdb/test/griffin/AllTablePartitionsTest.java

```diff
@@ -0,0 +1,175 @@
+        });
+    }
+
+    @Test
+    public void testAllPartitionsFewTablesWithEmptyTables() throws Exception {
+
+        String expectedHeader = "index\tpartitionBy\tname\tminTimestamp\tmaxTimestamp\tnumRows\tdiskSize\tdiskSizeHuman\treadOnly\tactive\tattached\tdetached\tattachable\ttableName";
+
+        String tableName1 = "aTable1";
+        String expected1 = "\n0\tMONTH\t2023-01\t2023-01-02T00:00:00.000000Z\t2023-01-08T00:00:00.000000Z\t7\tSIZE\tHUMAN\tfalse\ttrue\ttrue\tfalse\tfalse\tTABLE";
+
+        String tableName2 = "aTable2";
+        String expected2 = "\n0\tDAY\t2023-01-02\t2023-01-02T00:00:00.000000Z\t2023-01-02T00:00:00.000000Z\t1\tSIZE\tHUMAN\tfalse\tfalse\ttrue\tfalse\tfalse\tTABLE\n" +
+                "1\tDAY\t2023-01-03\t2023-01-03T00:00:00.000000Z\t2023-01-03T00:00:00.000000Z\t1\tSIZE\tHUMAN\tfalse\tfalse\ttrue\tfalse\tfalse\tTABLE\n" +
+                "2\tDAY\t2023-01-04\t2023-01-04T00:00:00.000000Z\t2023-01-04T00:00:00.000000Z\t1\tSIZE\tHUMAN\tfalse\tfalse\ttrue\tfalse\tfalse\tTABLE\n" +
+                "3\tDAY\t2023-01-05\t2023-01-05T00:00:00.000000Z\t2023-01-05T00:00:00.000000Z\t1\tSIZE\tHUMAN\tfalse\tfalse\ttrue\tfalse\tfalse\tTABLE\n" +
+                "4\tDAY\t2023-01-06\t2023-01-06T00:00:00.000000Z\t2023-01-06T00:00:00.000000Z\t1\tSIZE\tHUMAN\tfalse\tfalse\ttrue\tfalse\tfalse\tTABLE\n" +
+                "5\tDAY\t2023-01-07\t2023-01-07T00:00:00.000000Z\t2023-01-07T00:00:00.000000Z\t1\tSIZE\tHUMAN\tfalse\tfalse\ttrue\tfalse\tfalse\tTABLE\n" +
+                "6\tDAY\t2023-01-08\t2023-01-08T00:00:00.000000Z\t2023-01-08T00:00:00.000000Z\t1\tSIZE\tHUMAN\tfalse\tfalse\ttrue\tfalse\tfalse\tTABLE\n" +
+                "7\tDAY\t2023-01-09\t2023-01-09T00:00:00.000000Z\t2023-01-09T00:00:00.000000Z\t1\tSIZE\tHUMAN\tfalse\tfalse\ttrue\tfalse\tfalse\tTABLE\n" +
+                "8\tDAY\t2023-01-10\t2023-01-10T00:00:00.000000Z\t2023-01-10T00:00:00.000000Z\t1\tSIZE\tHUMAN\tfalse\tfalse\ttrue\tfalse\tfalse\tTABLE\n" +
+                "9\tDAY\t2023-01-11\t2023-01-11T00:00:00.000000Z\t2023-01-11T00:00:00.000000Z\t1\tSIZE\tHUMAN\tfalse\ttrue\ttrue\tfalse\tfalse\tTABLE";
+
+        String tableName3 = "aTable3";
+        String tableName4 = "aTable4";
+        assertMemoryLeak(() -> {
+            createTable(tableName1, PartitionBy.MONTH, 7);
+            createTable(tableName3, PartitionBy.DAY, 0);
+            createTable(tableName2, PartitionBy.DAY, 10);
+            createTable(tableName4, PartitionBy.DAY, 0);
+            String finallyExpected1 = replaceSizeToMatchOS(expected1, tableName1).replaceAll("\n$", "");
+            String finallyExpected2 = replaceSizeToMatchOS(expected2, tableName2);
+            assertAllPartitionTables(expectedHeader + finallyExpected1 + finallyExpected2);
+        });
+    }
+
+    private String replaceSizeToMatchOS(String expected, String tableName) {
+        return ShowPartitionsTest.replaceSizeToMatchOS(expected, new Utf8String(configuration.getRoot()), tableName, engine);
+    }
+
```

Comment:
Why is assertQuery called in a separate thread ?

Reply:
because I copypasted too much code from another test :sweat: sorted